### DISCUSSION TITLE
React native 0.64 & firebas v5@latest

### DISCUSSION THREAD

**Discussion by volodymyrFilatenko** [invertase/react-native-firebase#5374] [Q&A] [answered]
Hello, 
I have app on react native 0.64.1. My app is using firebase. Firebase version 5.6. 
App shut down on last version react native when user click on notification.
I need update firebase to 6 version or firebase v5 is supporting on react native 0.64.1?
Thanks, Vova.

**Comment by mikehardy** [comment, marked as answer]
react-native-firebase v5 has been unsupported for quite some time, do not use it under any circumstances I think.
Update to current versions

**Comment by volodymyrFilatenko** [comment]
Okay, Thanks!

**Comment by volodymyrFilatenko** [comment]
Hello, I updated react native firebase to v6 and got errors:
ERROR  Invariant Violation: Native module cannot be null.
 ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect. 

Could you help me?

**Reply by mikehardy** [reply]
@volodymyrFilatenko sure! Here's a whole script that generates a working project from scratch - give it a shot and read the script / examine it's output and make sure you've done all the things https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh

**Comment by volodymyrFilatenko** [comment]
I updated firebase, but I have a problem with connection to my server
after await messaging().getToken(). I got network request failed.

**Reply by mikehardy** [reply]
Troubleshooting in a discussion is not typically productive, you probably want to open a new issue here if you continue to have problems.

You will probably want to make sure you have an updated google-services json and plist form the firebase console for your apps though if you are updating from a very old version, and you may need to authorize the device check API on the google cloud platform console as well, or your app won't be fully authorized to use firebase services I think. Re-reading the current documents for firebase integration from the official firebase sources will help.

**Comment by volodymyrFilatenko** [comment]
Hello, thanks. I have solved this issue, but I got next issue).
When I send push with data only to ios in foreground then messaging().onMessage didn't get message.
Could you help me?

**Reply by mikehardy** [reply]
I believe there is already an issue for this, have you searched the issue list?

**Comment by volodymyrFilatenko** [comment]
I tried other solution, but don't help. messaging().onMessage didn't work
in foreground ios.
    ***@***.***/analytics": "12.0.0",
    ***@***.***/app": "12.0.0",
    ***@***.***/firestore": "12.0.0",
    ***@***.***/messaging": "12.0.0",

**Reply by mikehardy** [reply]
You should post on that issue then. Splitting efforts to triage and fix things among an issue and a discussion is counterproductive. I'll respond on the issue only to keep things focused there.
